src/main/java/org/zalando/pazuzu/config/OAuthConfiguration.java

```diff
@@ -17,34 +19,74 @@
 import org.zalando.stups.oauth2.spring.server.TokenInfoResourceServerTokenServices;
 import org.zalando.stups.tokens.config.AccessTokensBeanProperties;
 
+import java.util.Iterator;
+import java.util.List;
+
 @Configuration
 @EnableResourceServer
-@Profile(value = "production")
+@Profile(value = {"production", "test_oauth"})
 public class OAuthConfiguration extends ResourceServerConfigurerAdapter {
 
     @Autowired
     private AccessTokensBeanProperties accessTokensBeanProperties;
 
+    private static final String scopeMatcher = "#oauth2.hasScope('uid')";
+    private String adminMatcher;
+
+    @Value("#{'${pazuzu-registry.admins}'.split(',')}")
+    private List<String> adminList;
+
     @Override
     public void configure(ResourceServerSecurityConfigurer resources) throws Exception {
         // here is the important part for stups-expression-handler
         resources.expressionHandler(new ExtendedOAuth2WebSecurityExpressionHandler());
     }
 
+    private String buildAdminUserMatcher() {
+        Iterator<String> iterator = adminList.iterator();
+        StringBuilder result = new StringBuilder();
+
+        result.append(scopeMatcher);
+
+        if (adminList != null && adminList.isEmpty() == false) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid unnecessary comparisons in boolean expressions](https://www.codacy.com/app/Pazuzu/pazuzu-registry/file/2725695107/issues/source?bid=3131965&fileBranchId=3348590#l51)